src/Infrastructure/Adapter/BdTopoRoadGeocoder.php

```diff
@@ -75,4 +76,77 @@ public function findDepartmentalRoads(string $search, string $administrator): ar
 
         return $departmentalRoads;
     }
+
+    public function findReferencePoint(
+        string $lineGeometry,
+        string $administrator,
+        string $roadNumber,
+        int $pointNumber,
+        int $abscissa,
+    ): Coordinates {
+        $abscisse = $this->findReferencePointAsbcisse($administrator, $roadNumber, $pointNumber, $abscissa);
+
+        try {
+            $rows = $this->bdtopoConnection->fetchAssociative(
+                '
+                    SELECT ST_AsGeoJSON(
+                        ST_LocateAlong(
+                            ST_AddMeasure(
+                                ST_LineMerge(:geom),
+                                0,
+                                ST_Length(
+                                    ST_Transform(
+                                        ST_GeomFromGeoJSON(:geom),
+                                        2154
+                                    )
+                                )
+                            ),
+                            :abscissa
+                        )
+                    ) as point
```

Comment:
Apparemment d'après https://gis.stackexchange.com/a/170828on peut simplifier le calcul de longueur en mètres en castant en `::geography` ?

Peut-être aussi ajouter un commentaire sur l'enjeu de convertir `:geom` dans un système qui permet de calculer sa longueur en mètres (et pas en angles ou autre unité)

```suggestion
                      SELECT ST_AsGeoJSON(
                        ST_LocateAlong(
                          ST_AddMeasure(
                            ST_LineMerge(:geom),
                            0,
                            -- Convert to meters
                            ST_Length(:geom::geography)
                          ),
                          :abscissa,
                        )
                      ) as point
```

Reply:
Le casting via `::geography` ne fonctionne pas, du moins, ne retourne aucun résultat. J'ai rajouté un commentaire pour expliquer la conversion en mètre.